storage/blockdevice/COMPONENT_QSPIF/source/QSPIFBlockDevice.cpp

```diff
@@ -1099,6 +1102,15 @@ int QSPIFBlockDevice::_handle_vendor_quirks()
             tr_debug("Applying quirks for ISSI");
             _num_status_registers = 1;
             break;
+        case 0x01:
+            if (vendor_device_ids[1] == 0x02 && vendor_device_ids[2] == 0x20) {
+                // On a Cypress S25FS512S flash chip, bit-1 of the register CR3NV is
+                // expected to be 1 but its actual value is 0 on manufactured hardware.
+                // In order for configuration detection to work, this chip needs to be
+                // specially handled.
+                _CR3NV_quirk = false;
```

Comment:
This looks like we have detected a problematic S25FS512S device. Should this be `_CR3NV_quirk = true;`? Where do we set `_CR3NV_quirk` to `true` if anywhere?

Reply:
Good catch, it should be true here.

Reply:
Done